client/src/components/Workflow/Invocation/Graph/InvocationGraph.vue

```diff
@@ -0,0 +1,327 @@
+<script setup lang="ts">
+import { library } from "@fortawesome/fontawesome-svg-core";
+import {
+    faArrowDown,
+    faChevronDown,
+    faChevronUp,
+    faSignInAlt,
+    faSitemap,
+    faTimes,
+} from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/vue-fontawesome";
+import { BAlert, BButton, BCard, BCardBody, BCardHeader } from "bootstrap-vue";
+import { computed, onUnmounted, ref, watch } from "vue";
+
+import { components } from "@/api/schema";
+import ExpandedItems from "@/components/History/Content/ExpandedItems";
+import { JobProvider } from "@/components/providers";
+import { useDatatypesMapper } from "@/composables/datatypesMapper";
+import { useInvocationGraph } from "@/composables/useInvocationGraph";
+import { useWorkflowStateStore } from "@/stores/workflowEditorStateStore";
+import { type Step } from "@/stores/workflowStepStore";
+import type { Workflow } from "@/stores/workflowStore";
+import { withPrefix } from "@/utils/redirect";
+
+import WorkflowInvocationSteps from "./WorkflowInvocationSteps.vue";
+import Heading from "@/components/Common/Heading.vue";
+import ExternalLink from "@/components/ExternalLink.vue";
+import JobInformation from "@/components/JobInformation/JobInformation.vue";
+import JobParameters from "@/components/JobParameters/JobParameters.vue";
+import LoadingSpan from "@/components/LoadingSpan.vue";
+import WorkflowGraph from "@/components/Workflow/Editor/WorkflowGraph.vue";
+
+library.add(faArrowDown, faChevronDown, faChevronUp, faSignInAlt, faSitemap, faTimes);
+
+interface Props {
+    /** The invocation to display */
+    invocation: components["schemas"]["WorkflowInvocationElementView"];
```

Comment:
One minor general refactoring recommendation about importing API models in the frontend:

Instead of using components["schemas"], try to import those types or interfaces from `@/api`. In particular, for this model, you can find it in `@/api/invocations`, or you can export new ones if they are not already defined. This approach offers several advantages IMHO, such as better readability and the ability to update the model name in one place if it changes in the backend for any reason. Also, within those modules, we define some additional interfaces that may be useful, either by combining several interfaces or by defining missing interfaces while we migrate more schema models in the backend.

Reply:
I have used (and added to) the api schemas instead. Thank you for spotting this!